doc/design/logs.md

```diff
@@ -0,0 +1,35 @@
+# Logs in SQLFlow
+## Motivation
+In order to know well about the runtime status of the SQLFlow job, we need to count the number of `TRAIN/PREDICT/EXPLAIN/NORMAL` tasks over a period of time. For example, using ELK Stack for log query and analysis‎. Generally, such statistics components are implemented by logs.
+
+## logging library
+Several statistics components use `logtail` to collect logs. The `logtail` demands a file. If such information is written to `stdout`, we should redirect the `stdout` to a local file, the file size will increase until the SQLFlow server terminates, so we need a rolling file to store the log. Besides, a well-formatted log is readable.
+We use [Logrus](https://github.com/sirupsen/logrus) for layout, combining with [https://github.com/natefinch/lumberjack] for rolling the log file.
+
+## Log Formatter
+Logs in their raw form are typically a text format with one event per line, i.e. :    
+`2020-03-10 10:00:14 level={Level} requestID={RequestID} user={UserID} event={event} msg={Metric, Result or Details}`
+
+RequestID: We use workflow ID as request ID if existed, or UUID instead. With the same RequestID, organizing these logs(events) in order can be reduced to a story;
+user: The one who submits the SQL;
+event: The process of the task, such as: parsing;
+msg: Details.
+
+## Log Categories
+1. Statistics log
+    Including traffic and performance information.
+    1. Count the requests;
```

Comment:
In my mind, it is the ELK's responsibility to count, but not the SQLFlow server itself. Am I correct?

Reply:
Yes.
ELK is responsible for aggregation.